### ISSUE TITLE
Ability to add query hinting to make use of pg_hint_plan

### ISSUE THREAD

**Issue by aijorgenson** [tpetry/laravel-postgresql-enhanced#80]
It would be nice if there was a way to prefix queries with hints for extensions. For example, an officially recognized extension ( https://www.postgresql.org/about/news/pg_hint_plan-v160-released-2712/ ) pg_hint_plan which allows you to modify which index is utilized for a query.

https://github.com/ossc-db/pg_hint_plan

Syntax looks like so:
```SQL
/*+ Something(my_table index_name) Something(my_other_table other_index_name) */
select * from my_table mt join my_other_table mot on mot.my_table_id = mt.id
```

**Comment by tpetry** [comment]
Thats a great idea 👍 Also had it on my list for some day.

Should it be one long string or more PHP like as an array?

```php
Model::query()
  ->hintplan('Something(my_table index_name) Something(my_other_table other_index_name)')
  ->....

Model::query()
  ->hintplan(['Something(my_table index_name)', 'Something(my_other_table other_index_name)'])
  ->....
```

Hmm, `hintplan` as a method looks/sounds strange.

**Comment by aijorgenson** [comment]
@tpetry I'm relatively new to the PostgresSQL world, are there other extensions that could make use of a comment before the query like that?

Maybe `->queryPrefix()` and then you can toss in whatever string you would like? Like 

```php
Model::query()
    ->queryPrefix('/*+ Something(my_table index_name) Something(my_other_table other_index_name) */')
    ->....
```

**Comment by tpetry** [comment]
Theres nothing else.

**Comment by aijorgenson** [comment]
`->hint` or `->hintPlan` both sound great then!

**Comment by tpetry** [comment]
Sure. I could allowe both options.

**Comment by tpetry** [comment]
I don‘t see a conflict there. But I‘ll test it.

**Comment by 0x15f** [comment]
I've given this some thought / local testing and also came up with a new use case. 

PlanetScale recently rolled out [Traffic Control for Postgres](https://planetscale.com/docs/postgres/traffic-control), and it supports "tags" for creating their resource policies. Arguably, that could be a separate feature here, but in my mind, Tags and Hints are both simply comments prepended to the query. 

---

## Hints

Each hint is a data transfer object with explicit targeting.

- **Single Relation:**
    `new IndexScan('products', 'idx_products_store')`
- **Multi-relation targeting (joins):**
    `new HashJoin(['products', 'variants'])`
- *** Order-sensitive targeting:**
    `new Leading(['products', 'variants', 'orders'])`

Hints are compiled globally but are scoped using relation names rather than query structure.
_Con: This requires defining each object and maintaining them but does let people not have to learn hint syntax_


## Tags

Tags provide query-level metadata for the following purposes:
* traffic routing (PlanetScale)
* observability / tracing
* experimentation

Examples:
```
new QueryTag('route', 'replica')
new QueryTag('feature', 'search')
```

## Compilation

Both tags and hints are compiled into separate comment blocks at the root of the query.

```sql
SELECT
  /* route=replica feature=search */
  /*+ 
      IndexScan(products idx_products_store)
      HashJoin(products variants)
      Leading(products variants orders)
  */
...
```

### Rules:
* Tags and hints are never merged.
* They are always emitted at the root SELECT statement.
* Hints rely on relation names for targeting, not query nesting.

**Comment by tpetry** [comment]
Do you have some docs about the Planetscale tags? I thought its a setting on the connection and not query. There‘s also sqlcommenter which adds debugging and tracing information to queries.

**Comment by 0x15f** [comment]
Yeah, here are docs - https://planetscale.com/docs/postgres/monitoring/query-tags#tags

I have not tried SQLCommenter yet, going to play with it today.

**Comment by tpetry** [comment]
Ah, so PlanetScale follows the SQLCommenter syntax. But that one is expected to be at the end, so both could be fullfilled by proper SQLCommenter support. Much software already parses it at the end for monitoring purposes.

Bts. pg_hint_plan-like logic is coming to PG19 [1]. So any implementation should just try to not use naming of the new PG19 feature.

[1] https://www.dbi-services.com/blog/postgresql-19-pg_plan_advice/
